main.tf

```diff
@@ -1,6 +1,111 @@
 resource "google_container_cluster" "primary" {
   name = "${var.name}"
-  zone = "${var.region}-${var.zones[0]}"
+
+  zone  = "${var.region}-${var.zones[0]}"
+  count = "${var.regional_cluster ||  var.nat_enabled ? 0 : 1 }"
+
+  min_master_version = "${var.min_master_version}"
+  enable_legacy_abac = false
+
+  network    = "${var.network == "" ? terraform.workspace : var.network}"
+  subnetwork = "${google_compute_subnetwork.nodes-subnet.self_link}"
+  project    = "${var.project}"
+
+  ip_allocation_policy {
+    cluster_secondary_range_name  = "${google_compute_subnetwork.nodes-subnet.secondary_ip_range.0.range_name}"
+    services_secondary_range_name = "${google_compute_subnetwork.nodes-subnet.secondary_ip_range.1.range_name}"
+  }
+
+  additional_zones = [
+    "${formatlist("%s-%s", var.region, slice(var.zones,1,length(var.zones)))}",
+  ]
+
+  lifecycle {
+    ignore_changes = ["subnetwork"]
+  }
+
+  initial_node_count       = 1
```

Comment:
Is this parameter important if `remove_default_node_pool` is set?

Reply:
https://www.terraform.io/docs/providers/google/r/container_node_pool.html <- please see first example:   
# We can't create a cluster with no node pool defined, but we want to only use
# separately managed node pools. So we create the smallest possible default
# node pool and immediately delete it.